### ISSUE TITLE
[Crash][Issue][Mod Incompattibility] Quality Assurance (and likely others)

### ISSUE THREAD

**Issue by BraveCaperCat2** [ClaudeMetz/FactoryPlanner#465]
### Problem Description

When loading a save game or creating a new one with AMS machines enabled and this mod installed and enabled, the game reports an error, and the save game cannot be loaded. 
This might be due to an issue on line 762 of /backend/handlers/generator.lua, where recipes used in rocket silos with no main product crash the game.

I have provided a (possible) fix to this issue in form of a pull request: #464

I am still waiting on a log from the original user who reported it here:
[Mod Portal Discussions tab for Quality Assurance.](https://mods.factorio.com/mod/QualityAssurance/discussion/6772cd3840f746f6d85d9729)

### Reproduction

1. Install Quality Assurance.
2. Install Factory Planner.
3. Turn on "Enable AMS Machines" in the mod options for Quality Assurance.
4. Load a save game or create a new one.
5. See error.

**Comment by ClaudeMetz** [comment]
Hello, sorry for taking so long to respond. Appreciate the PR! I however am not able to reproduce this crash. Any idea what that might be about?

**Comment by BraveCaperCat2** [comment]
> Hello, sorry for taking so long to respond. Appreciate the PR! I however am not able to reproduce this crash. Any idea what that might be about?

Nope. The person who sent the error to me *still* hasn't sent a log, but I was able to reproduce at the time and found an error somewhere around the line 762 that I mentioned. I don't remember exactly since it's been a while though.

For now, I'll close this issue until I can reproduce again. (if I can reproduce again, it might not be reproducible anymore)

**Comment by ClaudeMetz** [comment]
Alright thanks, sorry again about taking so long 😅

**Comment by BraveCaperCat2** [comment]
> Alright thanks, sorry again about taking so long 😅

No worries.
